src/clj_kondo/core.clj

```diff
@@ -181,39 +179,42 @@
              :allow-string-hooks (-> config :hooks :__dangerously-allow-string-hooks__)
              :debug debug}
         lang (or lang :clj)
+        ;; primary file analysis and initial lint
         _ (core-impl/process-files (if parallel
                                      (assoc ctx :parallel parallel)
                                      ctx) lint lang filename)
         ;;_ (prn (some-> analysis deref :java-class-usages))
         ;; _ (prn :used-nss @used-nss)
-        idacs (when-not skip-lint
+        idacs (when (or dependencies (not skip-lint) analysis)
```

Comment:
Why does analysis need this step again?

Reply:
`l/lint-var-usages` is a bit of a misnomer. what actually happens there is resolving vars from the cached data and then linting which happens based on that retrieved data. So it could be called `post-analysis` / `post-cache` or something. It's just the naming that is confusing, but the logic makes sense (imo).